File: src/Domain/NutritionalContent.cs
using Microsoft.EntityFrameworkCore;

namespace Domain;

[Owned]
public class NutritionalContent
{
    public double Protein { get; set; }
    public double Carbohydrate { get; set; }
    public double Fat { get; set; }
    public double Kcal { get; set; }


    public NutritionalContent(double protein, double carbohydrate, double fat, double kcal)
    {
        Protein = protein;
        Carbohydrate = carbohydrate;
        Fat = fat;
        Kcal = kcal;
    }
}

File: src/Client/src/Models/NutritionalContent.ts
import { ObjectAssignable } from "./ObjectAssignable.ts";

/**
 * Represents the nutritional content in 100 grams of the food item.
 */
export class NutritionalContent extends ObjectAssignable {
    protein: number;
    carbohydrate: number;
    fat: number;
    kCal: number;
    unit: "grams" | "ml";

    constructor() {
        super();
        this.protein = 0.0;
        this.carbohydrate = 0.0;
        this.fat = 0.0;
        this.kCal = 0.0;
        this.unit = "grams";
    }
}
